src/routes/api/product.routes.js

```diff
@@ -0,0 +1,12 @@
+import express from 'express';
+import {protectRoute} from '../../middlewares/auth.middleware';
+import { isSeller } from '../../middlewares/user.middleware';
+import { createProduct } from '../../controllers/product.controller';
+import productValidation, { validateCategory } from '../../validations/product/product.validation';
+import { addCategory } from '../../controllers/category.controller';
+import { checkIfCategoryExists } from '../../middlewares/product.middleware';
+
+const productRoutes=express.Router();
+productRoutes.post('/products',protectRoute,isSeller,productValidation,validateCategory,createProduct);
```

Comment:
- Why do you have `validate category`?
- I can't see middleware to check the duplicate product
- Missing a middleware to check if category exist